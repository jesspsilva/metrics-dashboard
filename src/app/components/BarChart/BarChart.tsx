import styled from "styled-components";

import { BarChart as Chart } from "@tremor/react";

const ChartWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 20px;

  .recharts-tooltip-wrapper .text-tremor-content {
    text-transform: capitalize;
  }
`;

const H2 = styled.h2`
  font-size: 1.125rem;
  font-weight: 500;
`;

export default function BarChart({
  data,
  onChange,
  title,
  categories,
  colors,
}: {
  data: BarChartData[];
  onChange: (v: string) => void;
  title?: string;
  categories: string[];
  colors?: string[];
}) {
  return (
    <ChartWrapper>
      {title && <H2>{title}</H2>}
      <Chart
        data={data}
        categories={categories}
        index="name"
        yAxisWidth={48}
        className="mt-6"
        onValueChange={(v) => onChange(v && v.name ? String(v.name) : "")}
        colors={colors}
      />
    </ChartWrapper>
  );
}
