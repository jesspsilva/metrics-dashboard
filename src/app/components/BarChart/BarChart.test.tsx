import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

// Added this mock because the ResponsiveContainer component from recharts
// is causing an error when running the tests
// Reference: https://github.com/recharts/recharts/issues/727
jest.mock("recharts", () => ({
  ...jest.requireActual("recharts"),
  ResponsiveContainer: jest
    .fn()
    .mockImplementation((props: any) => <div {...props} />),
}));

import BarChart from "./BarChart";

const data = [
  {
    name: "Total",
    "% value": "10",
  },
  {
    name: "Cleaning",
    "% value": "20",
  },
];

const title = "Bar Chart Title";
const categories = ["% value"];

describe("BarChart", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <BarChart
        data={data}
        title={title}
        onChange={() => {}}
        categories={categories}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe("when there's a title", () => {
    it("should display the title", () => {
      const { queryByText } = render(
        <BarChart
          data={data}
          title={title}
          onChange={() => {}}
          categories={categories}
        />,
      );

      expect(queryByText(title)).toBeInTheDocument();
    });
  });

  describe("when there's no title", () => {
    it("should not display the title", () => {
      const { queryByText } = render(
        <BarChart data={data} onChange={() => {}} categories={categories} />,
      );

      expect(queryByText(title)).not.toBeInTheDocument();
    });
  });
});
