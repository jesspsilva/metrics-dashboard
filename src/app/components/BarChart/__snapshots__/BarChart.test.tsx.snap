// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-b143385c-0 jNqabe"
  >
    <h2
      class="sc-b143385c-1 mKzdD"
    >
      Bar Chart Title
    </h2>
    <div
      class="w-full h-80 mt-6"
    >
      <div
        class="h-full w-full"
      />
    </div>
  </div>
</DocumentFragment>
`;
